pkg/config/connections.go

```diff
@@ -69,6 +71,17 @@ func (c *GoogleCloudPlatformConnection) GetCredentials() *google.Credentials {
 }
 
 func (c GoogleCloudPlatformConnection) MarshalJSON() ([]byte, error) {
+	// If ADC is enabled, don't read or include service account fields as they're not used
+	if c.UseApplicationDefaultCredentials {
+		return json.Marshal(map[string]string{
+			"name":                                c.Name,
+			"project_id":                          c.ProjectID,
+			"location":                            c.Location,
+			"use_application_default_credentials": "true",
```

Comment:
I suggest always having the same keys available in the json and not having a difference in structure